panic when resuming a session

When I was resuming a session, unfortunately I got a handshake failure, meanwhile, I got a panic.
```
NO_ERROR: Handshake did not complete in time
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x6d264d]

goroutine 1745 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.(*clientSessionCache).Put(0xc000f6ed70, 0xc0019a2420, 0x13, 0x0)
        github.com/lucas-clemente/quic-go@v0.12.0/internal/handshake/qtls.go:58 +0x2d
github.com/marten-seemann/qtls.(*Conn).clientHandshake.func1(0xc0009dfea8, 0xc0002d0700, 0xc0019a2420, 0x13)
        github.com/marten-seemann/qtls@v0.4.0/handshake_client.go:167 +0x77
github.com/marten-seemann/qtls.(*Conn).clientHandshake(0xc0002d0700, 0xa62cc0, 0xc00117ad90)
        github.com/marten-seemann/qtls@v0.4.0/handshake_client.go:178 +0x5f4
github.com/marten-seemann/qtls.(*Conn).Handshake(0xc0002d0700, 0x0, 0x0)
        github.com/marten-seemann/qtls@v0.4.0/conn.go:1413 +0x132
github.com/lucas-clemente/quic-go/internal/handshake.(*cryptoSetup).RunHandshake.func1(0xc00013c240, 0xc001c6e000, 0xc0006b1560)
        github.com/lucas-clemente/quic-go@v0.12.0/internal/handshake/crypto_setup.go:260 +0x6e
created by github.com/lucas-clemente/quic-go/internal/handshake.(*cryptoSetup).RunHandshake
        github.com/lucas-clemente/quic-go@v0.12.0/internal/handshake/crypto_setup.go:258 +0xb0
```

--- a/internal/handshake/qtls.go
+++ b/internal/handshake/qtls.go
@@ -51,6 +51,10 @@ func (c *clientSessionCache) Get(sessionKey string) (*qtls.ClientSessionState, b
 }
 
 func (c *clientSessionCache) Put(sessionKey string, cs *qtls.ClientSessionState) {
+	if cs == nil {
+		c.ClientSessionCache.Put(sessionKey, nil)
+		return
+	}
 	// qtls.ClientSessionState is identical to the tls.ClientSessionState.
 	// In order to allow users of quic-go to use a tls.Config,
 	// we need this workaround to use the ClientSessionCache.

--- a/internal/handshake/qtls_test.go
+++ b/internal/handshake/qtls_test.go
@@ -125,5 +125,14 @@ var _ = Describe("qtls.Config generation", func() {
 			_, ok := qtlsConf.ClientSessionCache.Get("raboof")
 			Expect(ok).To(BeTrue())
 		})
+
+		It("puts a nil session state", func() {
+			csc := NewMockClientSessionCache(mockCtrl)
+			tlsConf := &tls.Config{ClientSessionCache: csc}
+			qtlsConf := tlsConfigToQtlsConfig(tlsConf, nil, &mockExtensionHandler{})
+			// put something
+			csc.EXPECT().Put("foobar", nil)
+			qtlsConf.ClientSessionCache.Put("foobar", nil)
+		})
 	})
 })

